### PR TITLE
chore:(auth)!: removed the multi guard

### PR DESCRIPTION
**Pull request by C-Ivan-Bandilla**
### 🛠 Changes
Removed the multi guard configuration
Repurpose the ChooseGuard to RoutePrefix class

### 🧠 Rationale
This is for simpler management of roles and permission

### 🧪 Test
Employee access admin
https://drive.google.com/open?id=1Ai1-lPpqkYEc94I6XY4Kl9f0lqOpYuyf&usp=drive_fs

Logout working
https://drive.google.com/open?id=1Ajuixw2IZnVeT55deUf04Q2pvjSwzKrT&usp=drive_fs

2FA setting working
https://drive.google.com/open?id=1AqM0bF0liNLkr8TxgPi9DLZ_XUYRpjHN&usp=drive_fs

Logout other session working
https://drive.google.com/open?id=1Az1O9MS6c9keTaLnWj_3vkKttedtjOZV&usp=drive_fs

Dashboard Pulse acessible
https://drive.google.com/open?id=1B3yI38msTpBslmsEU_GB7FsUa7fChKJA&usp=drive_fs

Admin Logi working
https://drive.google.com/open?id=1BI3edFVFC_pjZylNqKkHvRJyPCRUQHzu&usp=drive_fs

### ✔️ Checklist
This is a set of criteria to ensure the correctness of your PR. Check each that applied.
- [x] Is `staging` the base branch of this PR?
- [x] Do your changes not reveal sensitive information, such as secrets, API keys, etc?
- [ ] Are there no erroneous console logs, debuggers, or leftover code in your changes?

### CODE REVIEW

File: app/Providers/FortifyServiceProvider.php [hunk 1]
```diff
@@ -34,30 +33,6 @@ class FortifyServiceProvider extends ServiceProvider
...
-        if (request()->is('admin/*')) {
-            config(
-                [
-                    'fortify.guard' => 'admin',
-                    'fortify.home' => '/admin/dashboard',
-                ]
-            );
-        }
-
-        config(['fortify.prefix' => ChooseGuard::getByReferrer()]);
```
Comment (Carl-Tabuso) [inline app/Providers/FortifyServiceProvider.php:54]: does removing this not affect the 2FA challenge form? Like for employee, the prefix shall be `appUrl/employee/two-factor-challenge`

Comment (C-Ivan-Bandilla) [reply app/Providers/FortifyServiceProvider.php:54]: how do I test that ? I enabled 2fa but logging does not prompt it?

Comment (Carl-Tabuso) [reply app/Providers/FortifyServiceProvider.php:54]: I have not wriltten a test for login to 2fa challenge form flow. What you can do is to enable the 2fa and manually test it on login screen, see if you'll be redirected to the 2fa view. Make sure to put this code back `config(['fortify.prefix' => RoutePrefix::getByReferrer()]);`  and i think it'll work just fine.

Comment (C-Ivan-Bandilla) [reply app/Providers/FortifyServiceProvider.php:54]: It doesnt redirect me, I also tried it on chore/hr/applications branch where multi guard isn't removed  it doesn't prompt me either
image
image
https://drive.google.com/file/d/1BJXDKHlwVfVwnohYCJHwuffhd3LAPwiB/view?usp=sharing

Comment (Carl-Tabuso) [reply app/Providers/FortifyServiceProvider.php:54]: i couldn't figure this out. I might have to open a new pr for this.

Comment (C-Ivan-Bandilla) [reply app/Providers/FortifyServiceProvider.php:54]: i only knew that this should be working when you mentioned it in this pr as when I test the pr on account profile it doesnt show 2fa too I thought the 2fa is for setup only that time

Comment (Carl-Tabuso) [reply app/Providers/FortifyServiceProvider.php:54]: i'd have to look for the last commit the 2fa challenge form was still working. I think i fucked up at some point, esp when i was sort of experimenting with update password. I might have accidentally modified smth.

Comment (Carl-Tabuso) [reply app/Providers/FortifyServiceProvider.php:54]: Managed to find the root cause, it was this change from `Auth::validate()` to `Auth::attempt()` on `Auth\Admin\Login`

**Comment by Carl-Tabuso** [comment]
Can i push here, i just add a lil bit of change to `AppServiceProvider` and `Login` files. 2FA Challenge form is now working and does redirect me if i provide a TOTP or a recovery code. Additionally, I also ensure not allowing

**Comment by Carl-Tabuso** [comment]
Do you think we can make the routing more stupidly simple? I don't think it's necessary to have three separate routes in our case anymore. We have permissions, and, policies/gates are always available at our disposal when it comes to app logical authorization.

**Comment by C-Ivan-Bandilla** [comment]
Does that mean also the login route is centralized also?

**Comment by Carl-Tabuso** [comment]
I suppose so?

**Comment by Carl-Tabuso** [comment]
did u accidentally delete the view templates for admin and employee login? What i meant is to have a single route file to intercept every http request, but we're not using a single view template for login. Admin and employee are not given the option to authenticate via different social providers (e.g: google, facebook). The `remember me` token is also for guest, applicant, and employee only.

> Does that mean also the login route is centralized also?

What i understood is we'll consolidate every http request for auth in `web.php`

**Comment by Carl-Tabuso** [comment]
sorry man, did i get you confused? Maybe we can talk about it and open a new pr for this matter before doing it right on the fly. If you want, you can revert to https://github.com/POWERLANE-TEAM/emplo-hrms/pull/73/commits/e72a38553ef41ed9e0357e56908ad79c6272eddc

**Comment by C-Ivan-Bandilla** [comment]
Wait ill soft reset

**Comment by C-Ivan-Bandilla** [comment]
How about make it clear on f2f? I might skip first class tho

**Comment by Carl-Tabuso** [comment]
sure thing.

File: database/seeders/ApplicantSeeder.php [hunk 2]
```diff
@@ -49,8 +49,7 @@ function createApplicants($chunkStart, $chunk, $permissions)
                         'application_status_id' => ApplicationStatus::APPROVED,
                     ]);
 
-                    $role = Role::firstOrCreate(['name' => UserRole::BASIC]);
-                    $applicant_user->assignRole($role);
+                    $applicant_user->assignRole(UserRole::BASIC);
```
Comment (Carl-Tabuso) [inline database/seeders/ApplicantSeeder.php:52]: roles are reserved for employees

**Comment by C-Ivan-Bandilla** [comment]
> I suppose so? URL prefixing is quite easy here. We just have to wrap it in a conditional statement based on role and prefix it as such:
> 
> ```
> // route file
> 
> if (UserRole::ADVANCED) {
>     Route::prefix('advanced')->group(function { ... });
> }
> ```

I think on route file the auth helper returns null even the user is logged in 
image


https://stackoverflow.com/questions/38240330/access-authuser-in-route-php#:~:text=This%20is%20because%20when%20registering%20the%20routes%2C%20the%20middleware%20hasn%27t%20yet%20run%20so%20Auth%3A%3Acheck()%20is%20always%20going%20to%20be%20false%20because%20Laravel%20hasn%27t%20authenticated%20the%20user%20at%20that%20point.

**Pull request merged**